cluster/cmd/resource/resource.go

```diff
@@ -0,0 +1,395 @@
+	p.OperationStatus = handler.Success
+	p.Message = "Complete"
+	return p, nil
+}
+
+// Create handles the Create event from the Cloudformation service.
+func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey.Value(), *currentModel.ApiKeys.PrivateKey.Value())
+	if err != nil {
+		return handler.ProgressEvent{}, err
+	}
+
+	if _, ok := req.CallbackContext["stateName"]; ok {
+		return validateProgress(client, req, currentModel, "IDLE", "CREATING")
+	}
+
+	projectID := *currentModel.ProjectID.Value()
+
+	if len(currentModel.ReplicationSpecs) > 0 {
+		if currentModel.ClusterType.Value() != nil {
+			return handler.ProgressEvent{}, errors.New("ClusterType should be set when `ReplicationSpecs` is set")
+		}
+
+		if currentModel.NumShards.Value() != nil {
+			return handler.ProgressEvent{}, errors.New("NumShards should be set when `ReplicationSpecs` is set")
+		}
+	}
+
+	autoScaling := mongodbatlas.AutoScaling{
+		DiskGBEnabled: currentModel.AutoScaling.DiskGBEnabled.Value(),
+	}
+
+	clusterRequest := &mongodbatlas.Cluster{
+		Name:                     cast.ToString(currentModel.Name.Value()),
+		EncryptionAtRestProvider: cast.ToString(currentModel.EncryptionAtRestProvider.Value()),
+		ClusterType:              cast.ToString(currentModel.ClusterType.Value()),
+		BackupEnabled:            currentModel.BackupEnabled.Value(),
+		DiskSizeGB:               currentModel.DiskSizeGB.Value(),
+		ProviderBackupEnabled:    currentModel.ProviderBackupEnabled.Value(),
+		AutoScaling:              autoScaling,
+		BiConnector:              expandBiConnector(currentModel.BiConenctor),
+		ProviderSettings:         expandProviderSettings(currentModel.ProviderSettings),
+		ReplicationSpecs:         expandReplicationSpecs(currentModel.ReplicationSpecs),
+		ReplicationFactor:        currentModel.ReplicationFactor.Value(),
+		NumShards:                currentModel.NumShards.Value(),
+	}
+
+	if currentModel.MongoDBMajorVersion.Value() != nil {
+		clusterRequest.MongoDBMajorVersion = formatMongoDBMajorVersion(*currentModel.MongoDBMajorVersion.Value())
+	}
+
+	cluster, _, err := client.Clusters.Create(context.Background(), projectID, clusterRequest)
+	if err != nil {
+		return handler.ProgressEvent{}, fmt.Errorf("error creating cluster: %s", err)
+	}
+
+	log.Printf("Cluster %+v", cluster)
+
+	currentModel.ID = encoding.NewString(cluster.ID)
+	currentModel.StateName = encoding.NewString(cluster.StateName)
+
+	return handler.ProgressEvent{
+		OperationStatus:      handler.InProgress,
+		Message:              fmt.Sprintf("Create Cluster `%s`", cluster.StateName),
+		ResourceModel:        currentModel,
+		CallbackDelaySeconds: 65,
+		CallbackContext: map[string]interface{}{
+			"stateName": cluster.StateName,
+		},
+	}, nil
+}
+
+func clusterIsReady(client *mongodbatlas.Client, projectID, clusterName, targetState string) (bool, string, error) {
```

Comment:
We should rename this method something like "isClusterInTargetState" to reflect what it actually does, as it can check for more than just ready

Reply:
bump!